### PR TITLE
Move stock keycodes for Massdrop Alt & Ctrl to keyboard level

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

The original PRs that added the Alt and Ctrl keyboards to QMK included some keycodes that were included in the Massdrop-authored keymaps. They've since multiplied to other keymaps. This PR moves all the logic for these stock Massdrop keycodes to keyboard level, so they can be leveraged by QMK Configurator.

## Types of Changes

- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
The following users have keymaps edited by this PR:

* @abishalom
* @ash0x0
* @bontakun
* @daniel5151
* @elijahblake81
* @elisiano
* @emptyflask
* @ewersp
* @favorable-mutation
* @MatthewRobo
* @r-pufky
* @reywood
* @urbanvanilla
* @valen214
* @XScorpion2

There shouldn't be any changes in behaviour from this PR.

**Comment by r-pufky** [comment]
Thanks for doing this -- I do have a question though.

Shouldn't you be pulling in all of the default keycodes into the base? See `keyboards/massdrop/ctrl/keymaps/default_md/keymap.c`

The defaults I see pulled in currently are from `keyboards/massdrop/ctrl/keymaps/default/keymap.c`

I believe `default_md` adds rgb_matrix support for better QMK support, and the `default` keymap is the partially working QMK firmware from the original import (or please correct/explain me if I am wrong).

**Review by XScorpion2 — APPROVED** [review]

File: keyboards/massdrop/alt/alt.h [hunk 1]
```diff
@@ -34,3 +34,21 @@
...
+    U_T_AGCR,               // USB Toggle Automatic GCR control
+    DBG_TOG,                // DEBUG Toggle On / Off
+    DBG_MTRX,               // DEBUG Toggle Matrix Prints
+    DBG_KBD,                // DEBUG Toggle Keyboard Prints
+    DBG_MOU,                // DEBUG Toggle Mouse Prints
+    MD_BOOT,                // Restart into bootloader after hold timeout
+    NEW_SAFE_RANGE
+};
+
+#undef SAFE_RANGE
```
Comment (fauxpark) [inline keyboards/massdrop/alt/alt.h:53]: Does this work? `SAFE_RANGE` is not actually a define, it's an enum entry.

Comment (drashna) [reply keyboards/massdrop/alt/alt.h:53]: good point.  I personally put a note in the keymap.c file about it.

File: keyboards/massdrop/alt/alt.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "alt.h"
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    static uint32_t key_timer;
+
+    switch (keycode) {
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:22]:
```suggestion
    if (!process_record_user(keycode, record) { return false; }

    switch (keycode) {
```

File: keyboards/massdrop/alt/alt.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            if (record->event.pressed) {
+                key_timer = timer_read32();
+            } else {
+                if (timer_elapsed32(key_timer) >= 500) {
+                    reset_keyboard();
+                }
+            }
+            return false;
+        default:
+            return true; //Process all other keycodes normally
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:63]: This will prevent user keycodes from working.
```suggestion
```

File: keyboards/massdrop/alt/alt.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    return process_record_user(keycode, record);
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:66]:
```suggestion
    return true;
```

File: keyboards/massdrop/alt/alt.c [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "alt.h"
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    static uint32_t key_timer;
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:20]:
```suggestion
```

File: keyboards/massdrop/alt/alt.c [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            if (record->event.pressed) {
+                TOGGLE_FLAG_AND_PRINT(debug_keyboard, "Debug keyboard");
+            }
+            return false;
+        case DBG_MOU:
+            if (record->event.pressed) {
+                TOGGLE_FLAG_AND_PRINT(debug_mouse, "Debug mouse");
+            }
+            return false;
+        case MD_BOOT:
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:53]:
```suggestion
        case MD_BOOT:
        {
             static uint32_t key_timer;
```

File: keyboards/massdrop/alt/alt.c [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            if (record->event.pressed) {
+                key_timer = timer_read32();
+            } else {
+                if (timer_elapsed32(key_timer) >= 500) {
+                    reset_keyboard();
+                }
+            }
+            return false;
```
Comment (drashna) [inline keyboards/massdrop/alt/alt.c:61]:
```suggestion
            return false;
        }
```

**Pull request closed without merging**